versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/DatabaseAdapter.java

```diff
@@ -323,4 +323,35 @@ void updateRepositoryDescription(Function<RepoDescription, RepoDescription> upda
    * @param offset initial reflog id to read from
    */
   Stream<RefLog> refLog(Hash offset) throws RefLogNotFoundException;
+
+  /**
+   * Retrieve the binary objects identified via {@code keys}. Binary objects that do not exist are
+   * not returned.
+   */
+  Stream<BinaryContent> getBinaries(Stream<BinaryKey> keys);
+
+  /**
+   * Consistent put-binary operation.
+   *
+   * <p>Either a "put-if-absent", if {@code expectedHash} is empty or a compare-and-swap based on
+   * the value of {@code expectedHash}. * *
+   *
+   * <p><b>IMPORTANT</b> keys of the {@link BinaryContent} binary objects must not be used via this
+   * method and {@link #putBinaries(Stream)}.
```

Comment:
I'm not sure I understand this message, sorry :)

Reply:
Updated it.